### DISCUSSION TITLE
Calendar entries backup

### DISCUSSION THREAD

**Discussion by stelkintatko** [seedvault-app/seedvault#216] [Q&A]
Greetings to all of you!
wanted to see a backup of all calendar entries currently found in the phone. Possibly, might be a great feature, to be able to select, which accounts (sources) of calendar records to backup. Of course - not much sense to backup something like exchange, so it would be duplicated after restore. Please consider it... Thank you.

**Comment by grote** [comment]
AFAIK it isn't possible to store calenders locally. If you use a sync calender backend like exchange or dav x5, the entries should get synced and not backed up. If you use a special app that allows you to store events locally, this app should take care of backup.

**Reply by aereaux** [reply]
Sorry if this is not the right place to ask this, but it seems relevant.  My understanding is that lineageos (18.1) uses a version of Etar Calendar (installed as a system app?) which allows storing events offline.  Is there any way to back up these events?  I don't see Etar or anything relevant in the list of backed up apps.

Thanks.

**Reply by grote** [reply]
I don't know how they do this, but then it would be up to Etar or to whom is modifying Etar to add a backup functionality. All Seedvault does is backing up apps via the system's mechanism. It can't go into apps and grab their data on its own.

**Reply by aereaux** [reply]
Thanks for the quick response.  What could be the reason Etar doesn't show up in the list of backed up apps?  Is it because it's a system app or because it doesn't use the system backup mechanism?

**Reply by grote** [reply]
System apps are not shown, but that doesn't mean that it doesn't get backed up.

**Reply by aereaux** [reply]
OK, thanks.  I checked my decrypted backup and there is indeed an org.lineageos.etar entry.  I can't find anything for stored events, but that may be because I don't currently have any.  Either way, it'll be an Etar issue.

**Comment by andrzej-gmt** [comment]
Hi All,
I have just started to use SeedVault on LineageOS 18. I'm quite surprised, that it doesn't backup calendars. I don't think it is a part of Etar data. I guess these data belong to "Calendar Storage", which is a system app, probably similar to "Phone and Messaging Storage". I would expect calendars to be included in "System apps" backup, just like SMS or contacts.

**Reply by grote** [reply]
See above please.

**Comment by andrzej-gmt** [comment]
Could you be more specific?

**Comment by andrzej-gmt** [comment]
Maybe my previous message wasn't clear. So to summarize:

Local calendars exist for many years in Android. There are tools, that support local calendars on F-Droid:
https://f-droid.org/packages/org.sufficientlysecure.localcalendar/
https://f-droid.org/packages/org.sufficientlysecure.ical/

Etar can create local calendars too, but calendar data doesn't belong to Etar. You can clear all Etar data and local calendar with its events will remain untouched. You can see these events in Google Calendar too.

Local calendar data belongs to app com.android.providers.calendar. If you clear these data, then local calendars disappear. This is system app. 

My request is: could you consider to include support for com.android.providers.calendar in SeedVault?

**Reply by grote** [reply]
IMHO it is the tools (that add support for local calendars) responsibility to make the calendars available to the system backup. 

For contacts, no such tool is required, so we made a [separate app from seedvault](https://github.com/seedvault-app/seedvault/tree/android13/contactsbackup) whose only purpose is to make the addressbook available to the system backup.

**Comment by andrzej-gmt** [comment]
The 2 tools on F-Droid are equivalent to your app for contacts. I have used them to supplement SeedVault, when migrating from gapps to microg.
